packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -103,32 +129,49 @@ const pushTransaction = async (request: Request<MessageTypes.PushTransaction>) =
     const rawTx = request.payload.startsWith('0x') ? request.payload.slice(2) : request.payload;
     const api = await request.connect();
 
-    const { lastValidBlockHeight } = await api.getLatestBlockhash('finalized');
+    const {
+        value: { lastValidBlockHeight },
+    } = await api.rpc.getLatestBlockhash({ commitment: 'finalized' }).send();
 
-    const txBuffer = Buffer.from(rawTx, 'hex');
+    const txByteArray = getBase16Encoder().encode(rawTx);
+    const transaction = getTransactionDecoder().decode(txByteArray);
+    assertTransactionIsFullySigned(transaction);
 
-    try {
-        await api.sendRawTransaction(txBuffer, {
-            skipPreflight: true,
-            maxRetries: 0,
-        });
+    const compiledMessage = getCompiledTransactionMessageDecoder().decode(transaction.messageBytes);
+    const message = decompileTransactionMessage(compiledMessage);
+    if (isDurableNonceTransaction(message)) {
+        // TODO: Handle durable nonce transactions.
+        throw new Error('Unimplemented: Confirming durable nonce transactions');
+    }
+    const transactionWithBlockhashLifetime = transaction as typeof transaction &
+        TransactionWithBlockhashLifetime;
 
-        const signature = await api.sendRawTransaction(txBuffer, {
+    try {
+        const signature = getSignatureFromTransaction(transaction);
+        const sendAndConfirmTransaction = sendAndConfirmTransactionFactory(api);
+        // FIXME: If possible, it would be preferable to include the last valid block height for
+        // this transaction's blockhash in `MessageTypes.PushTransaction` rather than to fetch a
+        // fresh one above. The one fetched value does not represent the true expiry block height of
+        // the blockhash inside this transaction.
+        const transactionWithReconstitutedLastValidBlockHeight = {
+            ...transactionWithBlockhashLifetime,
+            lifetimeConstraint: {
+                ...transactionWithBlockhashLifetime.lifetimeConstraint,
+                lastValidBlockHeight,
+            },
+        };
+        await sendAndConfirmTransaction(transactionWithReconstitutedLastValidBlockHeight, {
+            commitment: 'confirmed',
+            maxRetries: 0n,
             skipPreflight: true,
-            maxRetries: 0,
         });
 
-        await confirmTransactionWithResubmit(api, txBuffer, signature, lastValidBlockHeight);
-
         return {
             type: RESPONSES.PUSH_TRANSACTION,
             payload: signature,
         } as const;
     } catch (error) {
-        if (
-            error instanceof SendTransactionError &&
-            error?.transactionError?.message === 'Internal error'
-        ) {
+        if (isSolanaError(error, SOLANA_ERROR__BLOCK_HEIGHT_EXCEEDED)) {
```

Comment:
The error system now offers errors that are scrutable by hundreds of codes, rather than having to match on the error message. Take a look and add errors as you need more granular semantics.